src/eth_provider/provider.rs

```diff
@@ -175,6 +181,7 @@ pub struct EthDataProvider<SP: starknet::providers::Provider> {
     database: Database,
     starknet_provider: SP,
     chain_id: u64,
+    account_balances: Arc<Mutex<HashMap<Address, (U256, u64)>>>,
```

Comment:
can I ask what is this?
it seems to be some in memory cache for balances?

why do we need it?

Reply:
Otherwise we need to query the balance for every retry which is useless. Also we can cache the balance since if you send one tx, high chance you will send a second one (approve, swap)

Reply:
But you delete the balance as soon as the tx is sent right?

Reply:
Yes, the balance is removed from the cache immediately after the `validate_transaction` is successfully executed

Reply:
hmmm I think maybe we should keep the balance, update it when a pending transaction is pruned (by substracting value + gas_used*gas_price) and prune accounts when we reach the limit of `ACCOUNT_BALANCES_DIMENSION` in a FIFO way, what do you think @eugypalu @tcoratger @Eikix

Reply:
Also I wonder if there is a way to avoid the Mutex on the `HashMap`, this will slow us down so much